Bottom: 64e3bc0ab2df6ca723e9a58edb11bf4da1de0ca0
Top:    6204ba6a52b3310828b66c548f5d1eee79cc16fa
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-04 21:26:53 +0000

Add support for drawing objects from the soldermask layer


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 0888e88..de4b8bb 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1328,7 +1328,8 @@ static void
 GhidDrawMask (int side, BoxType * screen)
 {
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
-
+  LayerType *Layer = LAYER_PTR (side == TOP_SIDE ? top_soldermask_layer : bottom_soldermask_layer);
+  struct poly_info info;
   OutputType *out = &Output;
 
   if (thin)
@@ -1342,6 +1343,14 @@ GhidDrawMask (int side, BoxType * screen)
     }
 
   hid_draw_use_mask (&ghid_graphics, HID_MASK_CLEAR);
+
+  info.layer = Layer;
+  info.drawn_area = screen;
+  r_search (Layer->polygon_tree, screen, NULL, poly_callback, &info);
+  r_search (Layer->line_tree, screen, NULL, line_callback, Layer);
+  r_search (Layer->arc_tree, screen, NULL, arc_callback, Layer);
+  r_search (Layer->text_tree, screen, NULL, text_callback, Layer);
+
   r_search (PCB->Data->pin_tree, screen, NULL, clearPin_callback_solid, NULL);
   r_search (PCB->Data->via_tree, screen, NULL, clearPin_callback_solid, NULL);
   r_search (PCB->Data->pad_tree, screen, NULL, clearPad_callback_solid, &side);
